clients/fides-js/src/fides.ts

```diff
@@ -0,0 +1,86 @@
+/***
+ * TODO: description
+ * TODO: preferably reconfigure tsup to generate fides.js instead of fides.global.js
+ * 
+ * Expected usage of this in an HTML file is:
+ * ```
+ * <script src="../dist/fides.global.js"></script>
+ * <script>
+ *   window.Fides.init({
+ *     consent: {
+ *       options: [...(default consent options here)]
+ *     }
+ *   });
+ * </script>
+ * ```
+ */
+import { gtm } from "./integrations/gtm";
+import { meta } from "./integrations/meta";
+import { shopify } from "./integrations/shopify";
+import { ConsentConfig } from "./lib/consent-config";
+import { getConsentContext } from "./lib/consent-context";
+import { CookieKeyConsent, getConsentCookie, makeDefaults } from "./lib/cookie";
+
+/**
+ * Configuration options for the fides.js script
+ */
+export interface FidesConfig {
+  consent: ConsentConfig;
+}
+
+type Fides = {
+  consent: CookieKeyConsent;
+  gtm: typeof gtm;
+  init: typeof init;
+  initialized: boolean;
+  meta: typeof meta;
+  shopify: typeof shopify;
+}
+
+declare global {
+  interface Window {
+    Fides: Fides;
+  }
+}
+
+/**
+ * Initialize the Fides object with configuration values
+ */
+const init = (config: FidesConfig) => {
+  console.debug("Initializing fides.js...");
+  // Configure the default values
+  const context = getConsentContext();
+  const defaults = makeDefaults({
+    config: config.consent,
+    context,
+  });
+
+  // Load any existing user preferences from the browser cookie
+  const consent = getConsentCookie(defaults);
+
+  // Initialize the window.Fides object
+  window.Fides.consent = consent;
+  window.Fides.initialized = true;
+}
+
+// TODO: come up with a clever-er way to do this? Seems janky!
+// Define the window.Fides object
+if (typeof window !== "undefined") {
```

Comment:
This might be fine, upon reflection. I'm not sure if there's really a "cleaner" way to make this module both safe to import into other code but also to use directly in a web app.

Technically the previous `fides-consent` package had two different entrypoints: the `lib/index.ts` which exported a bunch of library functions as a module, and the `fides-consent.js` file (this one) that was all bundled up...

I chose to try to make a single entrypoint because I felt that was much easier to understand. I doubt anyone really understood the magic that rollup was doing!

Reply:
I am all for the single entrypoint!